dev/WindowsAppRuntime_BootstrapDLL/MddBootstrap.cpp

```
@@ -337,10 +466,188 @@ CLSID FindDDLM(
+        {
+                // The package's Name prefix doesn't match the expected value. Skip it
+                continue;
+        }
+        if (packageNameSuffixLength > 0)
+        {
+            const auto offsetToSuffix{ packageNameLength - packageNameSuffixLength };
+            if (CompareStringOrdinal(packageName.c_str() + offsetToSuffix, static_cast<int>(packageNameSuffixLength), packageNameSuffix, static_cast<int>(packageNameSuffixLength), TRUE) != CSTR_EQUAL)
+            {
+                    // The package's Name suffix doesn't match the expected value. Skip it
+                    continue;
+            }
+        }
+        if (CompareStringOrdinal(packageId.PublisherId().c_str(), -1, expectedPublisherId, -1, TRUE) != CSTR_EQUAL)
+        {
+            // The package's PublisherId doesn't match the expected value. Skip it
+            continue;
+        }
+
+        // Is this DDLM in the major.minor release?
+        //
+        // NOTE: Package.InstalledLocation.Path can be expensive as it has to create
+        //       a StorageFolder just to get the path as a string. We'd like to use
+        //       Package.EffectivePath but that didn't exist until 20H1 and we need
+        //       to work down to RS5. So instead we'll use GetPackagePathByFullName()
+        //       as that exists since Win81 (and can be significantly faster than
+        //       Package.InstalledLocation).
+        const auto packageFullName{ packageId.FullName() };
+        wil::unique_cotaskmem_string packagePathBufferDynamic;
+        uint32_t packagePathLength{};
+        const auto rc{ GetPackagePathByFullName(packageFullName.c_str(), &packagePathLength, nullptr) };
+        if (rc != ERROR_INSUFFICIENT_BUFFER)
+        {
+            THROW_HR_MSG(HRESULT_FROM_WIN32(rc), "Enumeration: %ls", packageFullName.c_str());
```

there are a couple throws in the loop, which would short-circuit any later success - should these be continues?

---

No. General rule of thumb, error out on errors.

Possible outcomes some line of code:

* Success
* Failure
* Error

Success + Failure (e.g. not found) are expected cases we handle
Error (out-of-memory and others) terminate processing and return to caller. Errors aren't swallowed/ignored; trying to proceed could do more harm than good.

There's (pardon the pun) exceptions to these rules but that's the general practice.